From 4d8d28aa8e6a14090b3b54cd3fcfd10f8334bdd9 Mon Sep 17 00:00:00 2001
Message-Id: <4d8d28aa8e6a14090b3b54cd3fcfd10f8334bdd9.1650636481.git.keller@fs-net.de>
In-Reply-To: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
References: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
From: Hong Shi <hongsh@codeaurora.org>
Date: Mon, 29 Jun 2020 14:18:06 +0800
Subject: [PATCH 36/56] qcacld-2.0: Fix tx credit leak

It hit auth failure issue on some customer platform. Root cause is
credit leak when handle mgmt frame tx completion. When it get negative
value of credit delta, it didn't check but add the negative value to
target tx credit directly. Fix is update tx credits only if credit delta
is positive value.

Change-Id: I0767818c2fab9db3ad02e0f77ed7e0355e1e0ad5
CRs-Fixed: 2688270
---
 CORE/CLD_TXRX/HTT/htt_t2h.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CORE/CLD_TXRX/HTT/htt_t2h.c b/CORE/CLD_TXRX/HTT/htt_t2h.c
index 094de2d..6597a11 100644
--- a/CORE/CLD_TXRX/HTT/htt_t2h.c
+++ b/CORE/CLD_TXRX/HTT/htt_t2h.c
@@ -470,7 +470,7 @@ htt_t2h_lp_msg_handler(void *context, adf_nbuf_t htt_t2h_msg )
                     credit_delta = htt_tx_credit_update(pdev);
                     HTT_TX_MUTEX_RELEASE(&pdev->credit_mutex);
                 }
-                if (credit_delta) {
+                if (credit_delta > 0) {
                     ol_tx_target_credit_update(pdev->txrx_pdev, credit_delta);
                 }
             }
-- 
2.14.5

